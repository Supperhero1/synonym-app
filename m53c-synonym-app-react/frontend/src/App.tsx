import React from 'react';
import styles from './App.module.scss';
import InputWithButton from "./components/InputWithButton/InputWithButton";
import SynonymList from "./components/SynonymList/SynonymList";
import ErrorMessage from "./components/ErrorMessage/ErrorMessage";
import NoDataMessage from "./components/NoDataMessage/NoDataMessage";
import FlavorMessage from "./components/FlavorMessage/FlavorMessage";
import Footer from "./components/Footer/Footer"
import { useApp } from "./App.service";

function App({ forceConsistency = false }) {
    const service = useApp()

    return (
        <>
            <main className={styles.App}>
                <h1>THE SYNONYM APP</h1>
                <FlavorMessage dependency={service.wordToDisplay} forceConsistency={forceConsistency}/>
                <InputWithButton
                    buttonText={'Get synonyms'}
                    value={service.wordToCheck}
                    setValue={service.inputHandler}
                    onClick={service.fetchData}
                    id={'button1'}
                />
                <InputWithButton
                    buttonText={'Set new synonym'}
                    value={service.synonym}
                    setValue={service.synonymInputHandler}
                    onClick={service.sendNewSynonym}
                    id={'button2'}
                />
                {
                    service.wordToDisplay
                        ? <SynonymList
                            originalWord={service.wordToDisplay}
                            synonyms={service.data}
                            deleteHandler={service.deleteHandler}
                        />
                        : <NoDataMessage />
                }
                {service.error && <ErrorMessage message={service.error} />}
            </main>
            <Footer />
        </>
    );
}

export default App;
