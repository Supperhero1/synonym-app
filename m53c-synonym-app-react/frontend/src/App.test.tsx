import React from 'react';
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders title', () => {
    render(<App />);
    const linkElement = screen.getByText(/Synonym app/i as any);
    expect(linkElement).toBeInTheDocument();
});

it('renders correctly', () => {
    const tree = render(<App forceConsistency={true} />)

    expect(tree).toMatchSnapshot()
})
