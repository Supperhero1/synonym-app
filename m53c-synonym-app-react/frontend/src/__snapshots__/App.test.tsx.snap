// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="App"
      >
        <h1>
          THE SYNONYM APP
        </h1>
        <section
          class="flavorMessage"
        >
          <p>
            <i>
              substitute
               is 
              equivalent to 
              synonym
            </i>
          </p>
        </section>
        <div
          class="inputWithButtonWrapper"
        >
          <input
            value=""
          />
          <button>
            Get synonyms
          </button>
        </div>
        <div
          class="inputWithButtonWrapper"
        >
          <input
            value=""
          />
          <button>
            Set new synonym
          </button>
        </div>
        <h3>
          No word chosen! Type in a word and press "Get synonyms".
        </h3>
      </main>
      <footer
        class="footer"
      >
        <div>
          This site is powered by tests taking way too long to run on my monolith project
        </div>
        <div
          class="tooltip"
        >
          Nothing to see here, move along
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <main
      class="App"
    >
      <h1>
        THE SYNONYM APP
      </h1>
      <section
        class="flavorMessage"
      >
        <p>
          <i>
            substitute
             is 
            equivalent to 
            synonym
          </i>
        </p>
      </section>
      <div
        class="inputWithButtonWrapper"
      >
        <input
          value=""
        />
        <button>
          Get synonyms
        </button>
      </div>
      <div
        class="inputWithButtonWrapper"
      >
        <input
          value=""
        />
        <button>
          Set new synonym
        </button>
      </div>
      <h3>
        No word chosen! Type in a word and press "Get synonyms".
      </h3>
    </main>
    <footer
      class="footer"
    >
      <div>
        This site is powered by tests taking way too long to run on my monolith project
      </div>
      <div
        class="tooltip"
      >
        Nothing to see here, move along
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
